### PR TITLE
IGNITE-18540 Fixed processing of the default distribution zone.

### PR DESCRIPTION
**Pull request by sergeyuttsel**
https://issues.apache.org/jira/browse/IGNITE-18540

### CODE REVIEW

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java [hunk 1]
```diff
@@ -791,9 +813,19 @@ public boolean onUpdate(@NotNull WatchEvent evt) {
 
                             logicalTopology = newLogicalTopology;
 
-                            zonesConfiguration.distributionZones().value().namedListKeys()
+                            Stream.of(
```
Comment (sanpwc) [inline modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java:816]: I'd rather encapsulate all actions to be performed with zone  into some fooBar method and simply call fooBar for defaultZone and fooBar for all other zones from the list.

Comment (sergeyuttsel) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java]: Fixed, thanks.

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java [hunk 2]
```diff
@@ -289,54 +290,43 @@ public CompletableFuture<Void> alterZone(String name, DistributionZoneConfigurat
...
-                NamedListChange<DistributionZoneView, DistributionZoneChange> renameChange;
+            CompletableFuture<Void> change;
 
-                try {
-                    renameChange = zonesListChange.rename(name, distributionZoneCfg.name());
-                } catch (IllegalArgumentException e) {
-                    throw new DistributionZoneRenameException(name, distributionZoneCfg.name(), e);
-                }
+            if (DEFAULT_ZONE_NAME.equals(name)) {
+                change = zonesConfiguration.change(zonesChange -> zonesChange
```
[the remark this answers is not in the document]
Comment (sergeyuttsel) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java]: Fixed, thanks.

File: modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java [hunk 3]
```diff
@@ -590,6 +584,32 @@ private void updateMetaStorageOnZoneDelete(int zoneId, long revision) {
         }
     }
 
+    /**
+     * Updates {@link DistributionZoneChange} according to distribution zone configuration.
+     *
+     * @param zoneChange Zone change.
+     * @param distributionZoneCfg Distribution zone configuration.
+     */
+    private void updateZoneChange(DistributionZoneChange zoneChange, DistributionZoneConfigurationParameters distributionZoneCfg) {
```
[the remark this answers is not in the document]
Comment (sergeyuttsel) [reply modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java]: Fixed.

File: modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerTest.java [hunk 4]
```diff
@@ -226,59 +226,87 @@ public void testDropZoneIfNotExists() {
...
+    public void testUpdateNotDefaultZone() throws Exception {
+        testUpdateZone(ZONE_NAME);
+    }
 
-        assertNotNull(zone1, "Zone was not created.");
-        assertEquals(ZONE_NAME, zone1.name().value(), "Zone name is wrong.");
-        assertEquals(Integer.MAX_VALUE, zone1.dataNodesAutoAdjustScaleUp().value(), "dataNodesAutoAdjustScaleUp is wrong.");
-        assertEquals(Integer.MAX_VALUE, zone1.dataNodesAutoAdjustScaleDown().value(), "dataNodesAutoAdjustScaleDown is wrong.");
-        assertEquals(100, zone1.dataNodesAutoAdjust().value(), "dataNodesAutoAdjust is wrong.");
+    public void testUpdateZone(String zoneName) throws Exception {
```
[the remark this answers is not in the document]
Comment (sergeyuttsel) [reply modules/distribution-zones/src/test/java/org/apache/ignite/internal/distributionzones/DistributionZoneManagerTest.java]: Fixed.

**Review by sanpwc — APPROVED** [review]

**Pull request merged**